packages/common/src/eips/1559.json

```diff
@@ -0,0 +1,25 @@
+{
+    "name": "EIP-1559",
+    "number": 1559,
+    "comment": "Fee market change for ETH 1.0 chain",
+    "url": "https://eips.ethereum.org/EIPS/eip-1559",
+    "status": "Review",
+    "minimumHardfork": "london",
+    "gasConfig": {
+      "baseFeeMaxChangeDenominator": {
+        "v": 8,
+        "d": "Maximum base fee change denominator"
+      },
+      "elasticityMultiplier": {
+        "v": 2,
+        "d": "Maximum block gas target elasticity"
+      },
+      "initialBaseFee": {
+        "v": 1000000000,
+        "d": "Initial base fee on first EIP1559 block"
```

Comment:
The description from the EIP (in the code example) states for this base fee:

> # default to 1,000,000,000 for blocks before INITIAL_FORK_BLOCK_NUMBER

Does this match with description here respectively with the semantics (my beloved word for the day 😝 ) you applied this constant for in the code (didn't have a look yet)? I have to say I have my problems to interpret this comment from above myself, what does "before" mean in this context? Blocks before shouldn't have a base fee at all I guess? 🤔  Or is this read like "before it is implicitly always `initialBaseFee` and on the first EIP1559 block the base fee is actually calculated for the first time? Then this would still conflict with the comment from the code here saying - if I read correctly - that this fee is applied "as is" on the first block and there is not yet some calculation.

Just writing this so extensively here since I think this is important to have this really clear - also in conjunction with other clients - if the fee calculation starts at `EIP1159 block` or `EIP1159 block + 1.

Reply:
Yeah this is indeed confusing. It seems like the spec says that on the first block we should actually calculate the base fee and that it is thus not 1e9. But we actually assume that the first block has base fee 1e9 and this is also enforced in the block package. Since we synced to Aleut just fine this is weird and that could imply that other clients do not follow the spec (?). Will ask on R&D.

Reply:
this is the issue you mentioned jochem that the spec was updated to match all the clients' technically wrong but matching implementation, right? 😄 https://github.com/ethereum/EIPs/pull/3556